ipatests/test_integration/test_commands.py

```diff
@@ -679,3 +680,58 @@ def test_user_mod_change_capitalization_issue5879(self):
         assert 'Modified user "%s"' % (ipauser) in cmd.stdout_text
         assert 'First name: %s' % (modfirst) in cmd.stdout_text
         assert 'Last name: %s' % (modlast) in cmd.stdout_text
+
+    def test_sss_ssh_authorizedkeys(self):
+        """
+        When there is ssh key is added for IPA user then
+        'sss_ssh_authorizedkeys ipauser' should return ssh key.
+
+        Test for : https://pagure.io/SSSD/sssd/issue/3937
+
+        Steps:
+        1. add ipa user with ssh-key.
+        2. call sss_ssh_authorizedkeys and ssh key should be returned
+        """
+        user = 'testsshuser'
+        ssh_key_pair = tasks.generate_ssh_keypair()
+        pem_file = tempfile.mkdtemp(suffix='cert.pem', dir=paths.TMP)
+        key = tempfile.mkdtemp(suffix='private.key', dir=paths.TMP)
+        openssl_cmd = [
+            'openssl', 'req', '-x509', '-newkey', 'rsa:2048', '-days', '365',
+            '-nodes', '-keyout', key, '-out', pem_file, '-subj', '/CN=' + user]
+        self.master.run_command(openssl_cmd)
+        cert_b64 = self.get_cert_base64(self.master, pem_file)
+        try:
+            binary = '/usr/libexec/sssd/p11_child'
+            backup = tasks.FileBackup(self.master, binary)
+            content = '#!/bin/bash\nsleep 999999'
+            self.master.put_file_contents(binary, content)
+            tasks.kinit_admin(self.master)
+            password = 'Secret123'
+            testuser_password_confirmation = "%s\n%s\n" % (password,
+                                                           password)
+            self.master.run_command(['ipa', 'user-add', user,
+                                     '--first', user,
+                                     '--last', user,
+                                     '--password'],
+                                    stdin_text=testuser_password_confirmation)
+            cmd = 'ipa user-mod {} --ssh "{}"'.format(user, ssh_key_pair[1])
+            self.master.run_command(cmd)
+            self.master.run_command([
+                'ipa', 'user-add-cert', user, "--certificate=%s" % cert_b64])
+            self.master.run_command(['sss_cache', '-E'])
+            self.master.run_command(['getent', 'passwd', user])
+            result = self.master.run_command(
+                [paths.SSS_SSH_AUTHORIZEDKEYS, user])
+            assert ssh_key_pair[1] in result.stdout_text
+            # login to the system
+            client = paramiko.SSHClient()
+            client.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            client.connect(self.master.hostname,
+                           username=user,
+                           password=password)
+            client.close()
```

Comment:
You are establishing ssh connection from tests runner machine which in general case does not have sssd installed and is not enrolled to ipa domain.